File: editorial_ch/chap6-26_30/prob6-28.cpp
#include <string>
#include <iostream>
using namespace std;

// 用盧卡斯定理計算 ncr mod 3
int ncr(int x, int y) {
	if (x < 3 && y < 3) {
		if (x < y) return 0;
		if (x == 2 && y == 1) return 2;
		return 1;
	}
	return ncr(x / 3, y / 3) * ncr(x % 3, y % 3) % 3;
}

int main() {
	// 輸入
	int N; string C;
	cin >> N >> C;
	
	// 求出答案
	int answer = 0;
	for (int i = 0; i < N; i++) {
		int code = -1;
		if (C[i] == 'B') code = 0;
		if (C[i] == 'W') code = 1;
		if (C[i] == 'R') code = 2;
		answer += code * ncr(N - 1, i);
		answer %= 3;
	}
	
	// 將答案乘以 (-1)^(N-1) 
	if (N % 2 == 0) {
		answer = (3 - answer) % 3;
	}
	
	// 輸出答案（"BWR" 的第 answer 個字母）
	cout << "BWR"[answer] << endl;
	
	return 0;
}

File: editorial_ch/chap6-26_30/prob6-28.py
# 如果將本程式以 Python 提出，由於處理速度較慢，會導致超出執行時間限制（TLE）。
# 若以 PyPy3 提出，則能夠得到正確答案（AC）。

# 用盧卡斯定理計算 ncr mod 3
def ncr(n, r):
	if n < 3 and r < 3:
		A = [
			[ 1, 0, 0 ],
			[ 1, 1, 0 ],
			[ 1, 2, 1 ]
		]
		return A[n][r]
	return ncr(n // 3, r // 3) * ncr(n % 3, r % 3) % 3

# 輸入
N = int(input())
C = input()

# 求出答案
answer = 0
for i in range(N):
	code = "BWR".find(C[i])
	answer += code * ncr(N - 1, i)
	answer %= 3

# 將答案乘以 (-1)^(N-1) 
if N % 2 == 0:
	answer = (3 - answer) % 3

# 輸出答案（"BWR" 的第 answer 個字母）
print("BWR"[answer])